qiskit/qpy/binary_io/value.py

```diff
@@ -290,6 +290,29 @@ def _read_parameter_expression_v3(file_obj, vectors, use_symengine):
 
     payload = file_obj.read(data.expr_size)
     if use_symengine:
+        # This is a horrible hack to workaround the symengine version checking
+        # it's deserialization does. There were no changes to the serialization
+        # format between 0.11 and 0.13 but the deserializer checks that it can't
+        # load across a major or minor version boundary. This works around it
+        # by just lying about the generating version.
```

Comment:
I'm roughly concerned that Symengine 0.13 might have added new opcodes or something to the serialisation format, without that necessarily needing a format change.  Do we have any knowledge of whether that's the case or not?

Of course, the worst case would be that some opcodes change in a way that _doesn't_ fail the load.  I doubt we can fully determine that right now, and since the most pressing concern is to get single `Parameter` instances transmitting properly, I'm ok to move ahead, and revisit if we need to.

Reply:
I'm mildly optimistic that we should be ok here because we're not exposing the full set of symengine's capabilities and the pieces we do support are pretty basic to the library and likely didn't change much.

Skmming through https://github.com/symengine/symengine/blob/master/symengine/serialize-cereal.h I think it just uses the type names, I can't see how else it would work unless it uses the order the overload is added in the source file as an opcode (which we'd be fine on because there were not any substantial changes to that file). But I am not really familiar with the cereal library so this is just conjecture on my point.

Reply:
Oh yeah, I didn't really think much about how `ParameterExpression` limits the surface area we're exposed to.  That's definitely a positive. Let's move forwards with this - I'm not sure that me wringing my hands about it is going to get us any new knowledge, and in the meantime, this is hurting real deployments of Qiskit.